File: python/128_Longest_Consecutive_Sequence.py
#!/usr/bin/env python
# coding=utf-8

class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        set_nums = set(nums)
        
        longest = 0
        for num in nums:
            if num-1 not in set_nums:
                current_num = num
                length = 1
                
                while current_num+1 in set_nums:
                    current_num += 1
                    length += 1
                longest = max(length, longest)
                
        return longest
        

File: c++/128_Longest_Consecutive_Sequence.cpp
/*************************************************************************
	> File Name: c++/128_Longest_Consecutive_Sequence.cpp
	> Author: Pelhans 
	> Mail: me@pelhans.com
	> Created Time: 2018年03月28日 星期三 13时16分53秒
 ************************************************************************/

#include<iostream>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        unordered_map<int, bool> used;
        for (auto i : nums) used[i] = false;
        int longest = 0;
        for (auto i : nums){
            if (used[i] ) continue;
            int length = 1;
            used[i] = true;
            for (int j = i+1; used.find(j) != used.end(); ++j ){
                used[j] = true;
                ++length;
            }
            for (int j = i-1; used.find(j) != used.end(); --j ){
                used[j] = true;
                ++length;
            }
            longest = max(longest, length);
        }
        return longest;
    }
};
